.github/workflows/hatch_ci.yml

```diff
@@ -148,3 +173,66 @@ jobs:
           ls dist/*.tar.gz | xargs -n1 hatch run python -m twine check
           ls dist/*.whl | xargs -n1 hatch run python -m twine check
         shell: bash
+
+      - name: Run Whitesource Scan
+        if: ${{ 
+          github.repository_owner == 'SolaceDev' && 
+          inputs.whitesource_product_name != '' && 
+          inputs.whitesource_project_name != '' 
+          }}
+        uses: SolaceDev/solace-public-workflows/.github/actions/whitesource-scan@main
+        with:
+          whitesource_product_name: ${{ inputs.whitesource_product_name }}
+          whitesource_project_name: ${{ inputs.whitesource_project_name }}
+          whitesource_api_key: ${{ secrets.WHITESOURCE_API_KEY }}
+
+      - name: Run WhiteSource Policy Gate
+        if: ${{ 
+          github.repository_owner == 'SolaceDev' && 
+          inputs.whitesource_product_name != '' && 
+          inputs.whitesource_project_name != '' && 
+          github.ref == 'refs/heads/main' 
+          }}
+        uses: docker://ghcr.io/solacedev/maas-build-actions:latest
+        env:
+          WS_API_KEY: ${{ secrets.WHITESOURCE_API_KEY }}
+          WS_PRODUCT_NAME: ${{ inputs.whitesource_product_name }}
+          WS_PROJECT_NAME: ${{ inputs.whitesource_project_name }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          AWS_REGION: ${{ vars.MANIFEST_AWS_REGION }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        with:
+          entrypoint: /bin/sh
+          args: >
+            -c "
+            . $VIRTUAL_ENV/bin/activate &&
+            cd $ACTIONS_PATH/whitesource-policy-violation-checker &&
+            python whitesource_policy_violation_checker.py
+            "
+
+      - name: Run WhiteSource Vulnerability Gate
+        if: ${{ 
+          github.repository_owner == 'SolaceDev' && 
+          inputs.whitesource_product_name != '' && 
+          inputs.whitesource_project_name != '' && 
+          github.ref == 'refs/heads/main' 
+          }}
+        uses: docker://ghcr.io/solacedev/maas-build-actions:latest
+        continue-on-error: true
```

Comment:
Do not block right now. We will enforce this once the changes have been applied to all repos. Also, we need to communicate when the blocking will happen.